test-data/unit/check-newsemanal.test

```diff
@@ -3175,3 +3175,13 @@ class User:
     def __init__(self, name: str) -> None:
         self.name = name  # E: Cannot assign to a method \
                           # E: Incompatible types in assignment (expression has type "str", variable has type "Callable[..., Any]")
+
+[case testNewAnalyzerMemberNameMatchesTypeDict]
+from typing import Union, Any
+from typing_extensions import TypedDict
+
+class T(TypedDict):
+    b: b.T
+
+class b:
+    T: Union[Any]
```

Comment:
Could you please also check the `NamedTuple` case as I mentioned in the issue's comment to prevent regression on both cases

Reply:
Added a test for this